### PR TITLE
[alerts] Move alerts::storageItf to aos-core-sm-cpp

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: include/aos/common/alerts/alerts.hpp [hunk 1]
```diff
@@ -1,63 +0,0 @@
...
-    /**
-     * Destructor.
-     */
-    virtual ~StorageItf() = default;
-};
-
-/**
- * Sender interface.
- */
-class SenderItf {
```
Comment (mlohvynenko) [inline include/aos/common/alerts/alerts.hpp:45]: SendAlert interface is needed for AlertProcessing logic at least (https://github.com/aosedge/aos_core_lib_cpp/pull/184)

File: include/aos/common/alerts/sender.hpp [hunk 2]
```diff
@@ -4,41 +4,14 @@
...
-#define AOS_ALERTS_HPP_
+#ifndef AOS_ALERTS_SENDER_HPP_
+#define AOS_ALERTS_SENDER_HPP_
 
 #include "aos/common/cloudprotocol/alerts.hpp"
 #include "aos/common/types.hpp"
 
 namespace aos::alerts {
 
-/**
```
Comment (al1img) [inline include/aos/common/alerts/sender.hpp:15]: Let's keep this interface in `alerts.hpp` to be aligned with other modules.

Comment (mykola-kobets-epam) [reply include/aos/common/alerts/sender.hpp:15]: done

File: include/aos/common/alerts/alerts.hpp (hunk 1 above)
Comment (mykola-kobets-epam) [reply include/aos/common/alerts/alerts.hpp]: SenderItf is left here

**Review by al1img — APPROVED** [review]

**Pull request merged**